torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -238,6 +241,73 @@ std::string getNcclErrorDetailStr(
   return interpret + err;
 }
 
+control_plane::RegisterHandler dumpHandler{
+    "dump_nccl_trace_pickle",
+    [](const control_plane::Request& req, control_plane::Response& res) {
+      auto params = req.params();
+      auto includeCollectives = true;
+      auto includeStackTraces = true;
+      auto onlyActive = false;
+      size_t validParamCount = 0;
+
+      const auto& includeCollectivesIt = params.find("includecollectives");
+      if (includeCollectivesIt != params.end()) {
+        validParamCount++;
+        if (includeCollectivesIt->second == "false") {
+          std::cout << "includeCollectives: false" << std::endl;
+          includeCollectives = false;
+        } else if (includeCollectivesIt->second == "true") {
+          // do nothing since it's the default
+        } else {
+          res.setStatus(400);
+          res.setContent(
+              "Invalid value for " + includeCollectivesIt->first + ": " +
+                  includeCollectivesIt->second,
+              "text/plain");
+        }
+      }
```

Comment:
Maybe you want to make this logic a Macro so that people don't need to copy paste when adding new params?